lib/scsi_defs.h

```diff
@@ -125,5 +126,63 @@ namespace scsi_defs {
     ControlByte control_byte;
   } ABSL_ATTRIBUTE_PACKED; static_assert(sizeof(InquiryCommand) == kInquiryCommandSize);
 
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6 Table 58
+  // Fields are arranged according to Big Endian format
+  struct InquiryData {
+    uint8_t peripheral_qualifier: 3;
+    uint8_t peripheral_device_type: 5;
+    bool rmb: 1; // Removable Media Bit (RMB)
+    bool lu_cong: 1;
+    uint8_t reserved_1: 6;
+    uint8_t version: 8;
+    bool reserved_2: 2;
+    bool normaca: 1; //Normal ACA (NORMACA)
+    bool hisup: 1; // Hierarchical Support Bit (HISUP)
+    uint8_t response_data_format: 4;
+    uint8_t additional_length: 8;
+    bool sccs: 1; // SCC Supported
+    bool acc: 1; 
+    uint8_t tpgs: 2; // Target Port Group Support (TPGS)
+    bool third_party_copy: 1; // referred to as 3PC in the documentation
+    uint8_t reserved_3: 2;
+    uint8_t protect: 1;
+    bool obsolete_1: 1;
+    bool encserv: 1; // Enclosure Services Bit
+    bool vs_1: 1; // vendor specific bit
+    bool multip: 1; // multiple SCSI Port
+    bool obsolete_2: 1;
+    bool reserved_4: 2;
+    bool addr_16: 1; // SCSI 16-bit address support bit
+    bool obsolete_3: 1;
+    bool reserved_5: 1;
+    bool wbus_16: 1; // Wide Bus bit 
+    bool sync: 1;
+    bool obsolete_4: 1;
+    bool reserved_6: 1;
+    bool cmdque: 1; // Command Management Model bit
+    bool vs_2: 1; // vendor specific bit
+    uint8_t vendor_identification[8];
+    uint8_t product_identification[16];
+    uint8_t product_revision_level[4];
+    uint8_t vendor_specific_1[20];
+    uint8_t reserved_7: 4;
+    uint8_t clocking: 2;
+    bool qas: 1; // Quick Arbitration and Selection Supported bit
+    bool ius: 1; // Information Units Supported bit
+    uint8_t reserved_8: 8;
+    uint16_t vendor_descriptor_1;
+    uint16_t vendor_descriptor_2;
+    uint16_t vendor_descriptor_3;
+    uint16_t vendor_descriptor_4;
+    uint16_t vendor_descriptor_5;
+    uint16_t vendor_descriptor_6;
+    uint16_t vendor_descriptor_7;
+    uint16_t vendor_descriptor_8;
+    uint8_t reserved_9[22];
+    std::vector<uint8_t> vendor_specific; // this field has a variable size
```

Comment:
This field is of variable size. So, I made it a vector so that whenever this struct is initialized, this vector is built in the constructor. Let me know if you think there's a better way to do this

Reply:
I just realized that adding a vector is an unpacked non-POD field. This will make the compiler ignore the packed attribute.

Reply:
I think the way to go here is omit this field from this structure, and then in future parser logic take care to handle it specially.

Reply:
Sounds good! Will omit this field for now